src/net/tcp_stream.rs

```diff
@@ -171,6 +171,93 @@ impl TcpStream {
         let lunatic_error = LunaticError::from(id);
         Err(Error::new(ErrorKind::Other, lunatic_error))
     }
+
+    /// Sets write timeout for TcpStream
+    ///
+    /// This method will change the timeout for everyone holding a reference to the TcpStream
+    /// Once a timeout is set, it can be removed by sending `None`
+    pub fn set_write_timeout(&mut self, duration: Option<Duration>) -> Result<()> {
+        unsafe {
+            if let code @ 1.. = host::api::networking::set_write_timeout(
```

Comment:
I never saw this `if let _ @ 1..` syntax before and I write/read a lot of rust code. So I needed to look it up, but it was hard to google for (how do you even call this?) and I needed a few takes until I understood what it means. For simplicity reasons and not to scare away rust newbs I would prefer here:

```rust
let result = host::api::networking::set_write_timeout(..);
if result != 0 {
    let lunatic_error = LunaticError::from(code as u64);
    Err(Error::new(ErrorKind::Other, lunatic_error))
} else {
    Ok(())
}
```

Reply:
This code just binds the variable code to the value on the right hand side and makes sure it is always in the range of `1..<type>::MAX`. You can also use binding in match statements, sometimes useful when you need to match a few different arms, but keep the true value bound to a variable so that you don't have to repeat yourself very often. There's a chapter in the rust about doing this in match, but apparently not much about using it in if let statements
https://doc.rust-lang.org/rust-by-example/flow_control/match/binding.html

Anyway, I'll change it to something more common, no problem :)